### ISSUE TITLE
请教论文中一些没太看懂的地方

### ISSUE THREAD

**Issue by zjk000** [wzh9969/HPT#2]
作者你好，想请教论文中一些没太看懂的地方，望解答。
1、那个[PRED]token是一个新加入BERT原始词表的token吗？所以本文是对BERT的原始词表进行了扩充吗？[PRED]对应的embedding由[MASK]初始化得来，此后是否有更新？还是保持不变？
2、经过BERT的encode后得到[PRED]对应的hidden state（用的bert-base-uncased我没记错的话应该是一个768维的向量），此后是如何使用这个hidden state的我没太看懂。进一步，本文基于virtual label word的verbalizer是如何工作的我也没太看懂。具体的，常规的verbalizer是由[MASK]的hidden state经过MLM head得到词表中每一个token的概率，然后再取预先定义好的label word映射到对应的类别。那么本文由[PRED]的hidden state是如何得到virtual label word的，virtual label word又如何对应到相应的类别？
诚向作者请教以上本人在阅读论文时没太看懂的问题，谢谢！

**Comment by wzh9969** [comment]
1 是的，我们扩展了BERT的词表。[PRED]的表示会随着训练更新。
2 我们的工作方式和常规方式相同。MLM head预测[PRED]位置上词表中每一个token的概率，只不过这里的词表是virtual label word的集合，他们的embedding由BERT embedding中的一些词初始化，单独训练。

**Comment by zjk000** [comment]
你好，读了下代码，还是有一些没看懂的地方
```
self.cls = BertOnlyMLMHead(config)
...
prediction_scores = self.cls(sequence_output)
...
multiclass_logits = multiclass_logits[:, self.vocab_size:self.vocab_size + self.num_labels] + self.multiclass_bias
```
prediction_scores是由MLM head得到的话应该是一个 输入序列长度×词表长度 的向量，接下来的切片操作
`multiclass_logits[:, self.vocab_size:self.vocab_size + self.num_labels]`
我的理解是，对于序列中每个位置，仅保留后self.num_labels个结果，这self.num_labels个结果我是否可以理解为就是每个类别的分数？如果是的话，是否对BERT的原始词表还进行了除[PRED]以外的扩展？否则切片取
`self.vocab_size : self.vocab_size + self.num_labels`的话不是越界了吗?

**Comment by wzh9969** [comment]
是的，virtual label word就是扩充的部分

**Comment by zjk000** [comment]
好的，这下明白了，感谢解答！
